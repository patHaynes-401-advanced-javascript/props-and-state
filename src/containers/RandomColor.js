import React, { Component } from 'react';

export default class RandomColor extends Component {

  componentAddedToColor() {
    const colors = ['red', 'green', 'black', 'yellow', 'orange', 'purple', 'indigo'];
    window.setInterval(() => this.RandomColor(colors), 1000);
  }

  state = {
    color: 'grey',
    backgroundImage: ''
  }

  RandomColor(colors) {
    let colorDisplay = colors[Math.floor(Math.random() * colors.length)];
    if(colorDisplay === this.state.backgroundColor) {
      this.setState({ backgroundImage: 'url(https://i.dailymail.co.uk/i/pix/2016/03/18/15/324D202500000578-3498922-image-a-33_1458315465874.jpg)' });
    } else {
      this.setState({ backgroundColor: colorDisplay, backgroundImage: '' });
    }
  }

  render() {
    return (
      <div style={{
        backgroundColor: this.state.backgroundColor,
        backgroundImage: this.state.backgroundImage,
        width: '1420px',
        height: '1420px'
      }}></div>
    );
  }
}
