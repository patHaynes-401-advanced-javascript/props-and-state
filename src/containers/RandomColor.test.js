import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('RandomColor component', () => {
  it('renders RandomColor', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
