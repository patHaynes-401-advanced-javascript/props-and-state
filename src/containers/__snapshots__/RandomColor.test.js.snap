// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomColor component renders RandomColor 1`] = `
<div
  style={
    Object {
      "backgroundColor": undefined,
      "backgroundImage": "",
      "height": "1420px",
      "width": "1420px",
    }
  }
/>
`;
